src/test/components/credentialsForm/credentialsForm.test.js

```diff
@@ -55,51 +50,35 @@ describe('CredentialsForm', () => {
   });
 
   it('renders', async () => {
-    listSourceAuthentications = jest
-      .fn()
-      .mockImplementation(() => new Promise((resolve) => setTimeout(() => resolve(authentications), 100)));
+    listSourceAuthentications = jest.fn().mockResolvedValue(authentications);
 
     api.getSourcesApi = () => ({
       listSourceAuthentications,
     });
+    render(
+      componentWrapperIntl(
+        <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
+        store,
+        initialEntry
+      )
+    );
 
-    jest.useFakeTimers();
-
-    await act(async () => {
-      wrapper = mount(
-        componentWrapperIntl(
-          <Route path={routes.sourcesDetailEditCredentials.path} render={(...args) => <CredentialsForm {...args} />} />,
-          store,
-          initialEntry
-        )
-      );
-    });
-    wrapper.update();
-
-    expect(wrapper.find(Modal).props().title).toEqual('Edit account authorization credentials');
-    expect(wrapper.find(Bullseye)).toHaveLength(1);
-    expect(wrapper.find(Spinner)).toHaveLength(1);
-    expect(wrapper.find(SourcesFormRenderer)).toHaveLength(0);
-
-    await act(async () => {
-      jest.runAllTimers();
-    });
-    wrapper.update();
+    expect(screen.getByRole('progressbar')).toBeInTheDocument();
+    expect(screen.getByText('Edit account authorization credentials')).toBeInTheDocument();
 
-    expect(wrapper.find(Spinner)).toHaveLength(0);
-    expect(wrapper.find(Bullseye)).toHaveLength(0);
-    expect(wrapper.find(Modal).props().title).toEqual('Edit account authorization credentials');
-    expect(wrapper.find(Modal).props().description).toEqual(
-      'Use the fields below to reset your account authorization credentials. It may take some time to validate new information.'
-    );
-    expect(wrapper.find(Modal).props().isOpen).toEqual(true);
-    expect(wrapper.find(ModalFormTemplate)).toHaveLength(1);
-    expect(wrapper.find(SourcesFormRenderer).props().initialValues).toEqual({ authentication: authentications.data[1] });
-    expect(wrapper.find(SourcesFormRenderer).props().schema).toEqual({
-      fields: generateSuperKeyFields(sourceTypesData.data, AMAZON.name),
-    });
+    await waitFor(() => expect(() => screen.getByRole('progressbar')).toThrow());
 
-    jest.useRealTimers();
+    expect(screen.getByText('Edit account authorization credentials')).toBeInTheDocument();
+    expect(
+      screen.getByText(
+        'Use the fields below to reset your account authorization credentials. It may take some time to validate new information.'
+      )
+    ).toBeInTheDocument();
+
+    expect([...screen.getAllByRole('textbox')].map((e) => e.name || e.getAttribute('aria-label'))).toEqual([
+      'authentication.username',
+      'pre-filled-authentication',
+    ]);
```

Comment:
You could use `getByRole` and setting `name` option. Also these should be screen readable I don't think dotted and hyphened text are of any help.

Reply:
name is a form attribute, not sure how to make it screen readable. There are `<labels>` for that.

Reply:
However, using `name` would cause issues if there is a change: it will throw an error if it is changed.

This approach using `getAll` and `toEqual` will show a new results after a change (expected vs received), so it is far more developer friendly.

Reply:
I changed the aria label to be human readable. However, I guess the whole application would require some "blind" test to get everything correctly.

Reply:
Yup, this could work
```JS
screen.getByRole('textbox', { name: 'Access key ID' });
screen.getByRole('textbox', { name: 'Filled password' });
```

Reply:
Yeah, if there are any new elements in the form we won't break the test. Isn't there a selector that would fail if found anything else? I would prefer the `name` selector because it forces the devs to make the form screen readers compliant.

Reply:
Yeah, this would work almost the same just a little bit worse because it would throw errors in case of errors. I would rather want to see the changed value as with my previous approach. Or if you add a third element, this test will pass, etc.

Reply:
We can keep as is, it's already screen readable and has all the goodies. I just wanted to have it exposed. But you are correct on the need of updating test when label for `authentication.username` changes.